README.md

```diff
@@ -8,55 +10,44 @@ This buildpack will participate if all the following conditions are met
 
 * The `$BP_OPENTELEMETRY_ENABLED` is set to a truthy value (i.e. `true`, `t`, `1` ignoring case)
 
-The buildpack will do the following for Java applications:
-
-* Contributes the OpenTelemetry Java agent to a layer and configures `$JAVA_TOOL_OPTIONS` to use it
-
-## Testing
+At build time, the buildpack will do the following for Java applications:
 
-Given a Java application, you can package it by running the following command from the current folder.
+* Contributes the OpenTelemetry Java agent to a layer and configures `$JAVA_TOOL_OPTIONS` to use it.
+* By default, the agent is configured to be disabled (`OTEL_JAVAAGENT_ENABLED=false`).
```

Comment:
What is the rationale to have this disabled by default? If I set `BP_OPENTELEMETRY_ENABLED=true` there is a good probability I want the agent to be enabled. I'm OK leaving the metrics exporter disabled by default, that seems like extra so it should need to be turned on specifically.

Reply:
My assumption was that I probably want to use the same image build configuration across environments, even if in some of them don't have an OTel backend available.

For example, the development environment might not have an observability setup. In that case, it would be nice for developers not having to explicitly disable the agent so not to have continuous error logs about the OTel backend not being reachable.

For the environments where I have an observability setup, no matter what, I would have to configure the agent (service name, exporter url, token...), so adding an additional environment variable to enable the agent wouldn't feel like extra work.

What do you think? I'm open to both options.

Reply:
OK. That sounds totally reasonable to me. Thinking about the other APM agents and I think that's consistent as well. For example, Datadog has an env variable at build time to trigger the agent to be installed, but at runtime it's not enabled unless you configure the agent. Others are similar, but require a binding to be present at runtime to have the agent enabled. Let's leave it this way. Thanks for explaining.